app/main.py

```diff
@@ -1 +1,30 @@
-# write your code here
+class Animal:
+
+    alive = []
+
+    def __init__(self, name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self. health = health
+        self. name = name
+        self. hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    def bite(self, herbivore: Herbivore) -> None:
```

Comment:
that function input can be Herbivore, so type should be `Animal`
```suggestion
    def bite(self, animal: Animal) -> None:
```